### PR TITLE
Baremetal updates for walkthrough

### PR DESCRIPTION
**Pull request by AAkindele**
Update to Baremetal README for walkthrough.

- added optional instructions for setting up ssh key pair
- added optional instructions for adding a second node to the cluster
- added resource group prefix variable for folks using a shared subscription to avoid collisions.

https://github.com/retaildevcrews/ngsa/issues/186

### CODE REVIEW

File: 02-Bare-metal-setup/README.md [hunk 1]
```diff
@@ -27,23 +24,47 @@ az account set -s YourSubscriptionName
 
 > Optional - an SSH key is auto-generated when the codespace is created
 
-- Copy your local id_rsa and id_rsa.pub values to ~/.ssh
+- Copy your local id_rsa and id_rsa.pub values to ~/.ssh. This will allow you to ssh into the VM from your local machine if you lose access to the codespace.
 - `chmod 700 ~/.ssh/id_rsa*`
 
+If running locally, you can check for existing ssh keys. More info [here](https://docs.github.com/en/github-ae@latest/github/authenticating-to-github/checking-for-existing-ssh-keys).
+
+```bash
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md:32]: need the windows command too

dir %HOMEDRIVE%%HOMEPATH%\.ssh

File: 02-Bare-metal-setup/README.md [hunk 2]
```diff
@@ -27,23 +24,47 @@ az account set -s YourSubscriptionName
...
+
+ls -al ~/.ssh
+
+```
+
+If you dont have an existing public or private key pair, generate a new one. More info [here](https://docs.github.com/en/github-ae@latest/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).
+
+```bash
+
+ssh-keygen -t ed25519 -C "your_email@example.com"
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md]: I think you need to cd ~/.ssh or specify the output directory, but not sure.

File: 02-Bare-metal-setup/README.md [hunk 3]
```diff
@@ -27,23 +24,47 @@ az account set -s YourSubscriptionName
...
+
+# Note: If you are using a legacy system that doesn't support the Ed25519 algorithm, use:
+# ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
+
+```
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md:48]: the -t didn't work for me on Windows - it created a file named ed25519 instead of id_rsa

File: 02-Bare-metal-setup/README.md (hunk 2 above)
Comment (bartr) [reply 02-Bare-metal-setup/README.md]: Windows defaults to the correct file name / directory

File: 02-Bare-metal-setup/README.md [hunk 4]
```diff
@@ -27,23 +24,47 @@ az account set -s YourSubscriptionName
...
+RG_PREFIX=""
+RG_NAME="${RG_PREFIX}k8s-qs-rg"
+
 # Create a resource group
-az group create -l westus2 -n k8s-qs-rg
+az group create -l westus2 -n $RG_NAME
 
 # Create an Ubuntu VM and install prerequisites
-az vm create -g k8s-qs-rg --admin-username codespace -n k8s-qs --size standard_d2s_v3 --nsg-rule SSH --image Canonical:UbuntuServer:18.04-LTS:latest --os-disk-size-gb 128 --custom-data startup.sh --query publicIpAddress -o tsv
+vm_ip=$(az vm create -g $RG_NAME --admin-username codespace -n k8s-qs --size standard_d2s_v3 --nsg-rule SSH --image Canonical:UbuntuServer:18.04-LTS:latest --os-disk-size-gb 128 --custom-data startup.sh --query publicIpAddress -o tsv)
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md]: This won't work on Windows

File: 02-Bare-metal-setup/README.md [hunk 5]
```diff
@@ -27,23 +24,47 @@ az account set -s YourSubscriptionName
...
 
-# This will output an IP address
+# Print the VM IP address
+echo $vm_ip
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md]: won't work on Windows

File: 02-Bare-metal-setup/README.md [hunk 6]
```diff
@@ -52,19 +73,24 @@ az vm create -g k8s-qs-rg --admin-username codespace -n k8s-qs --size standard_d
 ```bash
 
 # ssh into the VM
-ssh codespace@YourIPAddress
+ssh codespace@${vm_ip}
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md]: won't work on Windows

File: 02-Bare-metal-setup/README.md [hunk 7]
```diff
@@ -111,4 +147,27 @@ sed -e "s/{PIP}/${PIP}/g" metalLB.yaml | k apply -f -
 
 ## Setup app
 
-- app readme <app/README.md>
+- app readme [app/README.md](app/README.md)
```
Comment (bartr) [inline 02-Bare-metal-setup/README.md]: this renders as a dupe - I think we can delete the first app readme

File: 02-Bare-metal-setup/app/01-zone-config-debug.yaml [hunk 8]
```diff
@@ -6,5 +6,5 @@ metadata:
 data:
   Region: "Debug"
   Zone: "Debug"
-  l8rLog: "l8rdebug"
+  l8rLog: "lrdebug"
```
Comment (bartr) [inline 02-Bare-metal-setup/app/01-zone-config-debug.yaml]: l8r is the official abbreviation for loadrunner :)

File: 02-Bare-metal-setup/app/03-app-debug.yaml [hunk 9]
```diff
@@ -35,7 +35,6 @@ metadata:
   labels:
     app: ngsa
 spec:
-  clusterIP:
```
Comment (bartr) [inline 02-Bare-metal-setup/app/03-app-debug.yaml:38]: why did you remove the clusterIP?  I assume this is in the service?

File: 02-Bare-metal-setup/app/README.md [hunk 10]
```diff
@@ -11,14 +11,20 @@ cd app
 kubectl create secret generic ngsa-secrets \
   --from-literal=WorkspaceId=unused \
   --from-literal=SharedKey=unused
-  
+
 # display the secrets (base 64 encoded)
 kubectl get secret ngsa-secrets -o jsonpath='{.data}'
 
+# you can decode the values using the base64 command
+kubectl get secret ngsa-secrets -o jsonpath='{.data.SharedKey}' | base64 -d
```
Comment (bartr) [inline 02-Bare-metal-setup/app/README.md]: this won't work if you run kubectl on Windows

File: 02-Bare-metal-setup/README.md (hunk 2 above)
Comment (AAkindele) [reply 02-Bare-metal-setup/README.md]: just double checked and it defaults to the correct file name and directory in ubuntu as well.

File: 02-Bare-metal-setup/README.md (hunk 3 above)
Comment (AAkindele) [reply 02-Bare-metal-setup/README.md:48]: i'll update this to use the default rsa. i originally got this from the github ssh docs.

File: 02-Bare-metal-setup/app/03-app-debug.yaml (hunk 9 above)
Comment (AAkindele) [reply 02-Bare-metal-setup/app/03-app-debug.yaml:38]: yep. this is the service. it was showing up as a linter error for me in vscode. it is okay though because the cluster will add this back in with a value.

File: 02-Bare-metal-setup/README.md (hunk 3 above)
Comment (kforeverisback) [reply 02-Bare-metal-setup/README.md:48]: Both ssh-keygen command worked for me. For ed25519 I used `ssh-keygen.exe -t ed25519 -b 4096 -C "mekram@microsoft.com" -f C:\Users\mekram\Documents\ssh-key`.
I am using `OpenSSH_for_Windows_7.7p1, LibreSSL 2.6.5`
RSA encryption is default for OpenSSL. So just removing `-t ed25519` should work.
